### ISSUE TITLE
No HTML report is generated when using the parallel module

### ISSUE THREAD

**Issue by unode** [ngless-toolkit/ngless#133]
This can be seen when running for instance [the parallel test](https://github.com/ngless-toolkit/ngless/tree/master/tests/parallel) with `ngless --create-report -o report parallel.ngl`.

The reason is that generating the final HTML report would require `collect()`ing reports for each sample. This final step doesn't currently happen, even if only one sample is present in the `samplefile`.

Since this report would require aggregating several samples, its format also needs to be different from the single sample one.

This can be one of the goals of GSoC 2020 - `Improved reporting of results` project idea.

**Comment by nairsajjal** [comment]
@unode Yeah, this can be a potential fix, and a potential Goal, I would be looking into this issue asap, and communicate the further details.

**Comment by mettejoe** [comment]
Wouldn't it be a possibility to just generate one html file pr sample with the sample name in the name of the html file. That was what I was expecting to get when I was using --create report for a script using the parallel module
